File: leetcode/solutions/6002.PerformStringShifts/stringShift.ts
export default function stringShift(
  s: string,
  shift: [number, number][]
): string {
  let amount = 0;
  for (const [direction, amo] of shift) {
    if (direction === 0) {
      amount -= amo;
    } else {
      amount += amo;
    }
  }

  if (amount === 0) {
    return s;
  }

  amount %= s.length;

  if (amount < 0) {
    amount = -amount;
    return s.substring(amount) + s.substring(0, amount);
  } else {
    return s.substring(s.length - amount) + s.substring(0, s.length - amount);
  }
}

File: leetcode/solutions/6002.PerformStringShifts/stringShift.go
package solution

func stringShift(s string, shift [][]int) string {
	var amount int

	for _, row := range shift {
		if row[0] == 0 {
			amount -= row[1]
		} else {
			amount += row[1]
		}
	}

	r := []rune(s)
	amount %= len(r)

	switch {
	case amount > 0:
		return string(r[len(r)-amount:]) + string(r[:len(r)-amount])

	case amount < 0:
		amount = -amount
		return string(r[amount:]) + string(r[:amount])

	default:
		return s
	}
}
